src/lib/components/DepositFormTitle.js

```diff
@@ -5,14 +5,27 @@
 // React-Invenio-Deposit is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
 import React, { Component } from 'react';
-import { Header } from 'semantic-ui-react'
-
-export class DepositFormTitle extends Component {
+import { connect } from 'react-redux';
+import { Header } from 'semantic-ui-react';
 
+class DepositFormTitleComponent extends Component {
   render() {
-    const content = this.props.isPublished ? "Edit Upload" : "New Upload";
-    return (
-      <Header as='h1' icon='upload' content={content} />
-    );
+    let content = '';
+    if (!this.props.isPublished) {
+      content = this.props.isNewVersion ? 'New version' : 'New upload';
+    } else {
+      content = 'Edit upload';
+    }
+    return <Header as="h1" icon="upload" content={content} />;
   }
 }
+
+const mapStateToProps = (state) => ({
+  isPublished: state.deposit.record.is_published,
+  isNewVersion: state.deposit.record.versions?.index > 1,
```

Comment:
Same here. Also `versions.index` is guaranteed no? (we could do without `?.`)

Reply:
No, because in case of a new draft (never saved it before) we get only empty values :) I could check in other ways if version exists, but I find this feature more elegant for this 😅  But, I had in mind what we discussed in the past and I am making sure I am using it only if the value is not guaranteed!

Reply:
Good to know, thanks!